test/browser/step_definitions/F2FBackendValidation.step.js

```diff
@@ -65,12 +68,76 @@ Then(
     const decodedBody = JSON.parse(
       Buffer.from(rawBody.replace(/\W/g, ""), "base64url").toString()
     );
-    // Strength Score
-    expect(decodedBody.vc.evidence[0].strengthScore).to.equal(3);
-    // Validity Score
-    expect(decodedBody.vc.evidence[0].validityScore).to.equal(2);
-    // Verification Score
-    expect(decodedBody.vc.evidence[0].verificationScore).to.equal(3);
+    const yotiMockIdId = this.yotiSessionId.substr(
+      this.yotiSessionId.length - 4
+    );
+    if (vcResponseData) {
+      // Strength Score
+      const expecedStrengthScore = eval(
+        "vcResponseData.s" + yotiMockIdId + ".strengthScore"
+      );
+      if (expecedStrengthScore) {
+        try {
+          expect(decodedBody.vc.evidence[0].strengthScore).to.equal(
+            expecedStrengthScore
+          );
+        } catch (error) {
+          console.log(
+            `Error validating Strength Score for yotiMockId: ${yotiMockIdId}`,
+            error
+          );
+          return error;
+        }
+      } else {
+        throw new Error(
+          `No expected Strength Score for yotiMockId: ${yotiMockIdId}`
+        );
+      }
+      // Validity Score
+      const expecedValidityScore = eval(
+        "vcResponseData.s" + yotiMockIdId + ".validityScore"
+      );
+      if (expecedValidityScore) {
+        try {
+          expect(decodedBody.vc.evidence[0].validityScore).to.equal(
+            expecedValidityScore
+          );
+        } catch (error) {
+          console.log(
+            `Error validating Validity Score for yotiMockId: ${yotiMockIdId}`,
+            error
+          );
+          return error;
+        }
+      } else {
+        throw new Error(
+          `No expected Validity Score for yotiMockId: ${yotiMockIdId}`
+        );
+      }
+      // Verification Score
+      const expecedVerificationScore = eval(
+        "vcResponseData.s" + yotiMockIdId + ".verificationScore"
+      );
+      if (expecedVerificationScore) {
+        try {
+          expect(decodedBody.vc.evidence[0].verificationScore).to.equal(
+            expecedVerificationScore
+          );
+        } catch (error) {
+          console.log(
+            `Error validating Verification Score for yotiMockId: ${yotiMockIdId}`,
+            error
+          );
+          return error;
+        }
+      } else {
+        throw new Error(
+          `No expected Verification Score for yotiMockId: ${yotiMockIdId}`
+        );
+      }
+    } else {
+      throw new Error("Verifiable Credential Validation JSON not found");
```

Comment:
Do you think we could extract this into a few functions? Finding it a little difficult to follow

Reply:
Nice looks really good!